src/modules/asset/asset.ts

```diff
@@ -0,0 +1,20 @@
+import { APICaller } from "../api-caller";
+import { AssetUrl as AssetUrlModel } from "../../objects/asset-url";
+
+export class Asset {
+  #apiCaller: APICaller;
+
+  constructor(apiCaller: APICaller) {
+    this.#apiCaller = apiCaller;
+  }
+
+  async getAssetById(assetId: string, tokenGetParam?: any) {
+    const assetData: any = await this.#apiCaller.get(
+      `/asset-urls/${assetId}`,
+      tokenGetParam
+    );
+
+    const asset = new AssetUrlModel(assetData);
```

Comment:
this should be either Library Image/Video/RawFile or Task Image/Video/RawFile asset. `apiCaller` will follow the redirection and fetch the data from the redirect URL. We need to detect what type of asset data is returned and return that object.
We should use the standard [`Location`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Location) header for 302 http code in our api response. But we are using `X-Resource-Location`. then `apiCaller` can follow the redirection.